web/docs/data-model/operations/actions.md

```diff
@@ -182,27 +186,62 @@ export const markTaskAsDone: MarkTaskAsDone<Pick<Task, 'id'>, void> = (
 }
 ```
 
+<SuperjsonNote />
+
+#### Type support for Actions
+
 Wasp automatically generates the types `CreateTask` and `MarkTaskAsDone` based on the declarations in your Wasp file:
 
 - `CreateTask` is a generic type that Wasp automatically generated based on the Action declaration for `createTask`.
 - `MarkTaskAsDone` is a generic type that Wasp automatically generated based on the Action declaration for `markTaskAsDone`.
 
-You can use these types to specify the Action's input and output types.
+Use these types to type the Action's implementation.
+Annotating Actions is optional but very helpful.
+Doing so properly types the Action's context.
 
-The Action `createTask` expects to get an object of type `{ description: string }` and returns the newly created task (an object of type `Task`).
+In this case, TypeScript will know the `context.entities` object must include the `Task` entity.
+TypeScript also knows whether the `context` object includes user information (it depends on whether your Action uses auth).
 
-The Action `markTaskAsDone`, expects an object of type `{ id: number }` and doesn't return anything (i.e., its return type is `void`).
+The generated types are generic and accept two optional type arguments: `Input` and `Output`.
 
-We've derived most of the payload types from the type `Task`.
+1. `Input` - The argument (i.e., payload) received by the Action function.
```

Comment:
`i.e.` feels very advanced, maybe we could the simplify language, just say`The payload`